Tinder.BLL/Services/GenericService.cs

```diff
@@ -24,8 +25,9 @@ public virtual async Task<TModel> CreateAsync(TModel model, CancellationToken ca
 
         public virtual async Task<TModel> DeleteAsync(Guid id, CancellationToken cancellationToken)
         {
-            var entity = await _repository.DeleteByIdAsync(id, cancellationToken);
-            return _mapper.Map<TModel>(entity);
+            _ = await _repository.GetByIdAsync(id, cancellationToken) ?? throw new NotFoundException("Entity with this id doesn't exis");
+            var deletedEntity = await _repository.DeleteByIdAsync(id, cancellationToken);
+            return _mapper.Map<TModel>(deletedEntity);

```

Comment:
What is the purpose of returning deleted entity?

Reply:
there's no particular purpose

Reply:
probably, after I delete it, I'll use the entity in some way, i haven't decided yet